libraries/api/src/expressions.js

```diff
@@ -137,10 +137,13 @@ const render = (compiledTemplate, params) => {
   if (ctmpl instanceof Array) {
     return ctmpl.map((value, i) => {
       if (i % 2 === 1) {
-        if (typeof params[value] !== 'string' && typeof params[value] !== 'number') {
-          throw new Error(`Scope expression template expected parameter '${value}' to be a string or number`);
+        switch (typeof params[value]) {
```

Comment:
This used to guard against forgetting a param. The empty string substitution might work but feels dangerous to me. Do we have access to the `?` at the end of `/namespaces/:namespace?` here? We could only allow `undefined` if it is optional. Maybe another choice would be to subsitute a placeholder string or something? /me waves hands a bit

Also happy to be shown why this isn't actually a problem. We could just add the explanation to this as a comment if so.